src/supplemental/quic/quic_api.c

```diff
@@ -1075,12 +1080,18 @@ quic_pipe_close(void *qpipe, uint8_t *code)
 		log_warn("close the QUIC stream!");
 		MsQuic->StreamClose(qstrm->stream);
 	}
+
```

Comment:
We need a lock here. Or a data race would happen between pipe_close and pipe_send/pipe_recv on sendq/recvq.